src/include/miopen/graphapi/opgraph.h

```diff
@@ -0,0 +1,274 @@
+#pragma once
+
+#include <miopen/graphapi/graphapi_tensor.hpp>
+
+#include <algorithm>
+#include <vector>
+#include <unordered_map>
+
+#include <cassert>
+
+namespace miopen {
+
+namespace graphapi {
+
+namespace internal {
+template <typename C>
+bool contains(const C& container, const typename C::value_type& val) noexcept
+{
+    return std::find(container.cbegin(), container.cend(), val) != container.cend();
+}
+} // end namespace internal
+
+class OpGraphBuilder;
+class OpGraph;
+
+struct OpNode
+{
+
+    using Edge = std::pair<OpNode*, Tensor*>;
```

Comment:
```suggestion
struct OpNode
{
    virtual ~OpNode() = default;

    using Edge = std::pair<OpNode*, Tensor*>;
```
A polymorphic base should have a virtual destructor.

Reply:
done, in cpp file.